native-src/sync_root_interface/callbacks/FetchData/FetchData.cpp

```diff
@@ -1,384 +1,212 @@
-                    NULL,
-                    ctx.requiredOffset,
-                    ctx.requiredLength,
-                    STATUS_UNSUCCESSFUL);
-            } else {
-                UINT64 totalSize = static_cast<UINT64>(ctx.fileSize.QuadPart);
-                progress = static_cast<float>(ctx.lastReadOffset) / static_cast<float>(totalSize);
-                Utilities::ApplyTransferStateToFile(ctx.path.c_str(), 
-                                                    ctx.callbackInfo, 
-                                                    totalSize, 
-                                                    ctx.lastReadOffset);
-                ::Sleep(CHUNKDELAYMS); 
-            }
-        }
-    } catch (...) {
-        Logger::getInstance().log("Excepción en file_incremental_reading.", LogLevel::ERROR);
-        FileCopierWithProgress::TransferData(
-            ctx.connectionKey,
-            ctx.transferKey,
-            NULL,
-            ctx.requiredOffset,
-            ctx.requiredLength,
-            STATUS_UNSUCCESSFUL);
-    }
+    size_t completed = offset + length;
 
-    file.close();
-    ctx.lastSize = newSize;
-    return ctx.lastReadOffset;
-}
+    wprintf(L"Bytes completed: %d. Total bytes: %d. Required offset: %d. Required length: %d.\n",
+            completed,
+            ctx->fileSize.QuadPart,
+            ctx->requiredOffset.QuadPart,
+            ctx->requiredLength.QuadPart);
 
-static napi_value response_callback_fn_fetch_data(napi_env env, napi_callback_info info)
-{
-    Logger::getInstance().log("response_callback_fn_fetch_data called", LogLevel::DEBUG);
-
-    size_t argc = 3;
-    napi_value argv[3];
-    napi_status status = napi_get_cb_info(env, info, &argc, argv, nullptr, nullptr);
-    if (status != napi_ok) {
-        Logger::getInstance().log("Failed to get callback info", LogLevel::ERROR);
-        return create_response(env, true, 0); 
-    }
+    winrt::com_ptr<IShellItem2> shellItem;
+    winrt::com_ptr<IPropertyStore> propStoreVolatile;
 
-    if (argc < 2) {
-        Logger::getInstance().log("This function must receive at least two arguments", LogLevel::ERROR);
-        return create_response(env, true, 0);
-    }
+    winrt::check_hresult(SHCreateItemFromParsingName(ctx->path.c_str(), nullptr, __uuidof(shellItem), shellItem.put_void()));
 
-    napi_valuetype valueType;
-    status = napi_typeof(env, argv[0], &valueType);
-    if (status != napi_ok || valueType != napi_boolean) {
-        Logger::getInstance().log("First argument should be boolean", LogLevel::ERROR);
-        return create_response(env, true, 0);
-    }
+    winrt::check_hresult(
+        shellItem->GetPropertyStore(
+            GETPROPERTYSTOREFLAGS::GPS_READWRITE | GETPROPERTYSTOREFLAGS::GPS_VOLATILEPROPERTIESONLY,
+            __uuidof(propStoreVolatile),
+            propStoreVolatile.put_void()));
 
-    bool response = false;
-    napi_get_value_bool(env, argv[0], &response);
+    if (completed < ctx->fileSize.QuadPart)
```

Comment:
Should you throw if the file size is bigger than the expected? The else is catching both cases